### ISSUE TITLE
Accessibility issue on Text Field Component with type Password/Search

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#1526]
Description  
---
The Text Field Component `type="password"` has following accessibility issues:
- The toggle (show/hide) button must have an accessible name to be announced in screen readers
- The toggle (show/hide) button must announce state changes of password visibility

The Text Field Component `type="search"` has following accessibility issues:
- The submit button has no accessible name

To Reproduce
---
1. Go to Text Field Component type Password
2. Open Screen Reader
3. Tab with keyboard to focus toggle button
4. Button has no descriptive text and state is unclear

Expected behavior
---
- Show/hide Button must be labelled correctly and should announce state changes to screen readers
- Submit Button should announce what to submit

Further information
---
https://incl.ca/show-hide-password-accessibility-and-password-hints-tutorial/

Test Environment
---
Platform/Browser/Version: All, VoiceOver
